dali/operators/reader/nvdecoder/cuvideodecoder.cc

```diff
@@ -78,31 +77,36 @@ const char* GetVideoChromaFormatString(cudaVideoChromaFormat eChromaFormat) {
 
 }  // namespace
 
-CUVideoDecoder::CUVideoDecoder() : decoder_{0},
-                                   decoder_info_{}, initialized_{false},
-                                   max_height_{0}, max_width_{0} {
-}
-
 CUVideoDecoder::CUVideoDecoder(int max_height, int max_width, int additional_decode_surfaces)
-                              : decoder_{0}, decoder_info_{}, initialized_{false},
+                              : decoder_{0}, decoder_info_{}, caps_{}, initialized_{false},
                                 max_height_{max_height}, max_width_{max_width},
                                 additional_decode_surfaces_{additional_decode_surfaces} {
 }
 
+CUVideoDecoder::CUVideoDecoder() : CUVideoDecoder(0, 0, 0) {
+}
+
 CUVideoDecoder::CUVideoDecoder(CUvideodecoder decoder)
-    : decoder_{decoder}, decoder_info_{}, initialized_{true},
-      max_height_{0}, max_width_{0} {
+    : CUVideoDecoder(0, 0, 0) {
+  decoder_ = decoder;
 }
 
 CUVideoDecoder::~CUVideoDecoder() {
     if (initialized_) {
-        NVCUVID_CALL(cuvidDestroyDecoder(decoder_));
+        try {
+          NVCUVID_CALL(cuvidDestroyDecoder(decoder_));
+        } catch (...) {
+          // something went totally wrong, terminate
+          std::terminate();
+        }
     }
 }
 
 CUVideoDecoder::CUVideoDecoder(CUVideoDecoder&& other)
-    : decoder_{other.decoder_}, initialized_{other.initialized_},
-      max_height_{other.max_height_}, max_width_{other.max_width_} {
+    : decoder_{other.decoder_}, decoder_info_{other.decoder_info_},
+      caps_{other.caps_}, initialized_{other.initialized_},
+      max_height_{other.max_height_}, max_width_{other.max_width_},
+      additional_decode_surfaces_{other.additional_decode_surfaces_} {
     other.decoder_ = 0;
     other.initialized_ = false;
```

Comment:
This variable seems redundant with respect to `decoder_`.

Reply:
Done